packages/SwingSet/src/kernel/virtualObjectManager.js

```diff
@@ -448,8 +452,9 @@ export function makeVirtualObjectManager(
       const innerSelf = { vobjID, rawData: initialData };
       const initialRepresentative = makeRepresentative(innerSelf, true);
       const initialize = initialRepresentative.initialize;
+      delete initialRepresentative.initialize;
+      harden(initialRepresentative);
       if (initialize) {
-        delete initialRepresentative.initialize;
         initialize(...args);
       }
       delete initialData[initializationInProgress];
```

Comment:
Should `initializationInProgress` be a WeakSet rather than a symbol? As a symbol on the object, it can be manipulated by the `initialize` function.

Reply:
I suppose a WeakSet would work equally well, but since the symbol is known only to the virtual object manager and the property is non enumerable I don't think there's a way for the `initialize` function to manipulate it or even know about it.  Is there some JavaScript-fu that I'm ignorant of that would enable that?

Reply:
(Actually, that's an academic question: since the WeakSet version has less mechansim I'll go with that anyway.)

Reply:
> property is non enumerable I don't think there's a way for the initialize function to manipulate it or even know about it

***IMPORTANT SAFETY TIP***

`Reflect.getOwnKeys` gets the names of all own properties on the object, whether or not they are symbols, whether or not they are so-called enumerable. Anything you use as a property name on an object is discoverable by anything with access to that object.

The JavaScript terminology "enumerable" is only of concern to a narrow range of issues: `for-in` loops, `Object.{keys, values, entries}`, `Object.assign`, Object spread and Object rest `{...x}`, and JSON. Probably a handful of other things not on the top of my head. It does *not* mean these objects cannot be enumerated!

Reply:
RIght, don't cross the streams.